src/calib/calib.c

```diff
@@ -153,152 +152,151 @@ void crop_sens(const long dims[DIMS], complex float* ptr, bool soft, float crth,
  * 
  * Parameters:
- *  var         - Estimated variance in data.
- *  evec_dims   - The eigenvector dimensions. 
- *  evec_data   - The eigenvectors.
- *  eptr        - The eigenvalues.
- *  calreg_dims - Dimension of the calibration region.
- *  calreg      - Calibration data.
+ *	var		- Estimated variance in data.
+ *	evec_dims	- The eigenvector dimensions. 
+ *	evec_data	- The eigenvectors.
+ *	eptr		- The eigenvalues.
+ *	calreg_dims     - Dimension of the calibration region.
+ *	calreg	        - Calibration data.
  *
  */
 static float sure_crop(float var, const long evec_dims[5], complex float* evec_data, complex float* eptr, const long calreg_dims[4], const complex float* calreg) {
 
-    // Must be in ascending order.
-    float cvals[] = {0.7, 0.71, 0.72, 0.73, 0.74, 0.75, 0.76, 0.77, 0.78, 0.79, 
-                     0.8, 0.81, 0.82, 0.83, 0.84, 0.85, 0.86, 0.87, 0.88, 0.89,
-                     0.9, 0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99}; 
+	// Must be in ascending order.
+	float cvals[] = {0.7, 0.71, 0.72, 0.73, 0.74, 0.75, 0.76, 0.77, 0.78, 0.79, 
+		         0.8, 0.81, 0.82, 0.83, 0.84, 0.85, 0.86, 0.87, 0.88, 0.89,
+			 0.9, 0.91, 0.92, 0.93, 0.94, 0.95, 0.96, 0.97, 0.98, 0.99}; 
+
+	long num_cvals = sizeof(cvals)/sizeof(float);
 
-    long num_cvals = sizeof(cvals)/sizeof(float);
+	// Creating hanning window
+	complex float* h_calreg = md_alloc(4, calreg_dims, CFL_SIZE); 
+	apply_hanning(calreg_dims, calreg, h_calreg);
 
-    // Creating hanning window
-    complex float* h_calreg = md_alloc(4, calreg_dims, CFL_SIZE); 
-    apply_hanning(calreg_dims, calreg, h_calreg);
+	// Zero pad Image
+	long im_dims[5];
+	md_select_dims(5, 15, im_dims, evec_dims);
+	complex float* im = md_alloc(5, im_dims, CFL_SIZE);
+	md_resize_center(5, im_dims, im, calreg_dims, h_calreg, CFL_SIZE);
 
-    // Zero pad Image
-    long im_dims[5] = { evec_dims[0], evec_dims[1], evec_dims[2], evec_dims[3], 1 };
-    complex float* im = md_alloc(5, im_dims, CFL_SIZE);
-    md_resize_center(5, im_dims, im, calreg_dims, h_calreg, CFL_SIZE);
 
+	long im_dims_prod = 1;
+	for (int idx = 0; idx < 4; idx ++)
+		im_dims_prod *= im_dims[idx];
 
-    long im_dims_prod = 1;
-    for (int idx = 0; idx < 4; idx ++)
-        im_dims_prod *= im_dims[idx];
+	// Inverse Unitary FFT
+	fftscale(5, im_dims, FFT_FLAGS, im, im);
+	ifftc(5, im_dims, FFT_FLAGS, im, im);
 
-    // Inverse Unitary FFT
-    fftscale(5, im_dims, FFT_FLAGS, im, im);
-    ifftc(5, im_dims, FFT_FLAGS, im, im);
+	// Eigenvectors (M) 
+	complex float* M = md_alloc(5, evec_dims, CFL_SIZE);
+	md_copy(5, evec_dims, M, evec_data, CFL_SIZE);
 
-    // Eigenvectors (M) 
-    long M_dims[5] = { evec_dims[0], evec_dims[1], evec_dims[2], evec_dims[3], evec_dims[4] };
-    complex float* M = md_alloc(5, M_dims, CFL_SIZE);
-    md_copy(5, M_dims, M, evec_data, CFL_SIZE);
+	// Eigenvalues (W)
+	long W_dims[5];
+	md_select_dims(5, 23, W_dims, evec_dims);
```

Comment:
Since your function is "MRI-aware", you should use the dimensions and flags macros defined in `mri.h`.

For example, use `md_select_dims(5, ~COIL_FLAG, W_dims, evec_dims)`

Does that make sense?